lib/bark.ex

```diff
@@ -2,56 +2,75 @@ defmodule Bark do
   require Logger
 
   # Logs a list of kv pairs
-  @spec warn(any(), Keyword.t()) :: any()
+  @spec warn(any(), Keyword.t() | binary()) :: any()
   def warn(env, opts), do: Logger.warn(parse_message(env, opts))
 
-  @spec info(any(), Keyword.t()) :: any()
+  @spec info(any(), Keyword.t() | binary()) :: any()
   def info(env, opts), do: Logger.info(parse_message(env, opts))
 
-  @spec error(any(), Keyword.t()) :: any()
+  @spec error(any(), Keyword.t() | binary()) :: any()
   def error(env, opts), do: Logger.error(parse_message(env, opts))
 
-  @spec debug(any(), Keyword.t()) :: any()
+  @spec debug(any(), Keyword.t() | binary()) :: any()
   def debug(env, opts), do: Logger.debug(parse_message(env, opts))
 
-  defp parse_message(env, opts) do
-    {function_name, arity} =
-      env
-      |> Map.get(:function)
-      |> case do
-        {function_name, arity} -> {function_name, arity}
-        _ -> {"None", 0}
-      end
-
-    module =
-      env
-      |> Map.get(:module)
-      |> case do
-        nil -> "None"
-        module -> module |> Atom.to_string() |> String.replace("Elixir.", "")
-      end
-
-    Enum.reduce(
-      opts,
-      "module=#{module} command=#{function_name}/#{arity}",
-      fn {k, v}, acc ->
-        "#{acc} #{Atom.to_string(k)}=#{log_value(v)}"
-      end
-    )
+  defp parse_message(env, opts) when is_list(opts) do
+    env
+    |> add_caller_context(opts)
+    |> to_log_string()
   end
 
+  defp parse_message(env, opts) when is_binary(opts) do
+    env
+    |> add_caller_context(message: opts)
```

Comment:
When just a binary string is passed it gets set as the value for `message`, thoughts?

Reply:
Could encourage laziness, but I like this in some ways too.

Reply:
Yeah I went back and forth on this, however as a drop in replacement for Logger I think this is an overall improvement.

Reply:
I generally lean towards this encouraging behavior we don't want. Ideally, I don't think we almost ever use just a message, or do we want that?